packages/@ionic/cli/src/lib/project/vue/serve.ts

```diff
@@ -14,34 +14,112 @@ export class VueServeRunner extends ServeRunner<VueServeOptions> {
     return {};
   }
 
-  modifyOpenUrl(url: string, options: VueServeOptions): string {
+  modifyOpenUrl(url: string, _options: VueServeOptions): string {
     return url;
   }
 
   async serveProject(options: VueServeOptions): Promise<ServeDetails> {
-    const cli = this.getPkgManagerServeCLI();
-
-    if (!await cli.resolveScript()) {
-      throw new RunnerException(
-        `Cannot perform serve.\n` +
-        `Since you're using the ${chalk.bold('Vue')} project type, you must provide the ${chalk.green(cli.script)} npm script so the Ionic CLI can serve your project.`
-      );
-    }
-
-    const [ externalIP, availableInterfaces ] = await this.selectExternalIP(options);
+    const [externalIP, availableInterfaces] = await this.selectExternalIP(options);
 
     const port = options.port = await findClosestOpenPort(options.port);
 
-    await cli.serve(options);
+    const vueScripts = new VueServeCLI(this.e);
+    await vueScripts.serve(options);
 
     return {
-      custom: false,
-      protocol: 'http',
+      custom: vueScripts.resolvedProgram !== vueScripts.program,
+      protocol: options.https ? 'https' : 'http',
       localAddress: 'localhost',
       externalAddress: externalIP,
       externalNetworkInterfaces: availableInterfaces,
       port,
       externallyAccessible: ![BIND_ALL_ADDRESS, ...LOCAL_ADDRESSES].includes(externalIP),
     };
   }
+
+}
+
+export class VueServeCLI extends ServeCLI<VueServeOptions> {
+  readonly name = 'Vue CLI Service';
+  readonly pkg = '@vue/cli-service';
+  readonly program = 'vue-cli-service';
+  readonly prefix = 'vue-cli-service';
+  readonly script = SERVE_SCRIPT;
+  protected chunks = 0;
+
+  async serve(options: VueServeOptions): Promise<void> {
+    this.on('compile', chunks => {
+      if (chunks > 0) {
+        this.e.log.info(`... and ${strong(chunks.toString())} additional chunks`);
+      }
+    });
+
+    return super.serve(options);
+  }
+
+  protected stdoutFilter(line: string): boolean {
+    if (this.resolvedProgram !== this.program) {
+      return super.stdoutFilter(line);
+    }
+
+    const strippedLine = stripAnsi(line);
+    const compileMsgs = ['Compiled successfully', 'Compiled with warnings', 'Failed to compile'];
+    if (compileMsgs.some(msg => strippedLine.includes(msg))) {
+      this.emit('ready');
+      return false;
+    }
+
+    if (strippedLine.match(/.*chunk\s{\d+}.+/)) {
+      this.chunks++;
+      return false;
+    }
+
+    if (strippedLine.includes('Compiled successfully')) {
+      this.emit('compile', this.chunks);
+      this.chunks = 0;
+    }
+
+    // const endBannerMsgs = ['development build', 'production build']
+    // if (endBannerMsgs.some(msg => strippedLine.includes(msg))) {
+    //   return false;
+    // }
```

Comment:
do we still need this?

Reply:
Mmmm...I haven't made up my mind yet. If we keep it, it will ignore some output , but it could be valuable information. I'd say for let's keep it for now then get feedback.